skore/src/skore/_sklearn/_plot/metrics/precision_recall_curve.py

```diff
@@ -134,595 +125,29 @@ def __init__(
-
-        else:  # multiclass-classification
-            info_pos_label = None  # irrelevant for multiclass
-            labels = self.precision_recall["label"].cat.categories
-            colors = sample_mpl_colormap(
-                colormaps.get_cmap("tab10"),
-                10 if len(estimator_names) < 10 else len(estimator_names),
-            )
-            idx = 0
-
-            for est_idx, estimator_name in enumerate(estimator_names):
-                est_color = colors[est_idx]
-
-                for label_idx, label in enumerate(labels):
-                    query = "label == @label & estimator_name == @estimator_name"
-                    average_precision = self.average_precision.query(query)[
-                        "average_precision"
-                    ]
-
-                    precision_recall = self.precision_recall.query(query)
-
-                    for split_idx, segment in precision_recall.groupby(
-                        "split", observed=True
-                    ):
-                        if split_idx == 0:
-                            label_kwargs = {
-                                "label": (
-                                    f"{estimator_name} "
-                                    f"(AUC = {average_precision.mean():0.2f} "
-                                    f"+/- {average_precision.std():0.2f})"
-                                )
-                            }
-                        else:
-                            label_kwargs = {}
-
-                        line_kwargs["color"] = est_color
-                        line_kwargs["alpha"] = 0.6
-                        line_kwargs_validated = _validate_style_kwargs(
-                            line_kwargs, pr_curve_kwargs[idx]
-                        )
-
-                        (line,) = self.ax_[label_idx].plot(
-                            segment["recall"],
-                            segment["precision"],
-                            **(line_kwargs_validated | label_kwargs),
-                        )
-                        lines.append(line)
-
-                        idx = idx + 1
-
-                    info_pos_label = f"\n(Positive label: {label})"
-                    _set_axis_labels(self.ax_[label_idx], info_pos_label)
-
-            if self.data_source in ("train", "test"):
-                legend_title = f"{self.data_source.capitalize()} set"
-            else:
-                legend_title = "External set"
-
-            for ax in self.ax_:
-                _, labels = ax.get_legend_handles_labels()
-                ax.legend(loc="lower left", title=legend_title)
-
-            self.figure_.suptitle("Precision-Recall Curve")
-
-        return self.ax_, lines, info_pos_label
-
     @DisplayMixin.style_plot
     def plot(
         self,
         *,
-        estimator_name: str | None = None,
-        pr_curve_kwargs: dict[str, Any] | list[dict[str, Any]] | None = None,
+        subplot_by: str | Literal["auto"] | None = "auto",
```

Comment:
I called this one `subplots_by` in the coefficient PR. I'm not sure which one is best.